backend/src/database/dbService.ts

```diff
@@ -0,0 +1,25 @@
+import { promises as fs } from 'fs';
+import path from 'path';
+
+const dbFilePath = path.resolve('./src/database/db.json');
+
+interface Recipe {
+    id: string;
+    title: string;
+    ingredients: string;
+    steps: string;
+}
+
+interface DatabaseSchema {
```

Comment:
You can move the interfaces to a separate file, `dbSchema.d.ts` or `dbSchema.types.ts`